lib/fastlane/plugin/wpmreleasetoolkit/helper/git_helper.rb

```diff
@@ -13,6 +13,23 @@ def self.has_git_lfs
         `git config --get-regex lfs`.length > 0
       end
 
+      # Switch to the given branch and pull its latest commits.
+      #
+      # @param [String,Hash] branch Name of the branch to pull.
+      #        If you provide a Hash with a single key=>value pair, it will build the branch name as `"#{key}/#{value}"`,
+      #        i.e. `checkout_and_pull(release: version)` is equivalent to `checkout_and_pull("release/#{version}")`.    
+      #
+      # @return [Bool] True if it succeeded switching and pulling, false if there was an error during the switch or pull.
+      #
+      def self.checkout_and_pull(branch)
```

Comment:
I wonder if this would make more sense to have as `checkout_and_pull!` and crash if it doesn't work instead? 

I can't think of a lot of times where we'd want to carry on if this failed, but WDYT?

Reply:
Yeah, I mainly did that to mimick `commit(message:files:push:)` I think.

But in act we actually use that feature at some call sites – e.g. in `get_user_build_version` from `android_beta_build_prechecks.rb:53` – so it's still useful to have this return a `Bool` instead of crashing.